lib/inventory_sync/async/query_inventory_job.rb

```diff
@@ -4,32 +4,50 @@ module InventorySync
   module Async
     class QueryInventoryJob < ::Actions::EntryAction
       include ActiveSupport::Callbacks
-      include ::ForemanRhCloud::CloudAuth
+      include ::ForemanRhCloud::CertAuth
 
       define_callbacks :iteration, :step
 
+      def plan(organizations, **params)
+        actual_params = params.merge(
+          {
+            organization_ids: Array(organizations).map(&:id),
+          }
+        )
+
+        plan_self(actual_params)
+      end
+
       def run
         run_callbacks :iteration do
-          page = 1
-          loop do
-            api_response = query_inventory(page)
-            results = HostResult.new(api_response)
-            logger.debug("Downloaded cloud inventory data: #{results.percentage}%")
-
-            run_callbacks :step do
-              results
+          organizations.each do |organization|
+            unless cert_auth_available?(organization)
+              logger.debug("Subscription manifest not available, skipping #{action_name} for organization #{organization.name}")
+              next
             end
 
-            page += 1
-            break if results.last?
+            page = 1
+            loop do
+              api_response = query_inventory(organization, page)
```

Comment:
I am getting the error: `ActiveRecord::SubclassNotFound: Invalid single-table inheritance type: Nic::Managed is not a subclass of Nic::Base`

some logs:
```
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/inheritance.rb:241:in `find_sti_class'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/inheritance.rb:215:in `discriminate_class_for_record'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/persistence.rb:257:in `instantiate'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/querying.rb:58:in `block (2 levels) in find_by_sql'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/result.rb:62:in `block in each'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/result.rb:62:in `each'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/result.rb:62:in `each'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/querying.rb:58:in `map'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/querying.rb:58:in `block in find_by_sql'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activesupport-6.0.4.8/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/querying.rb:56:in `find_by_sql'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/bullet-7.0.1/lib/bullet/active_record60.rb:19:in `find_by_sql'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/relation.rb:824:in `block in exec_queries'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/relation.rb:842:in `skip_query_cache_if_necessary'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/relation.rb:811:in `exec_queries'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/association_relation.rb:36:in `exec_queries'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/relation.rb:626:in `load'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/relation.rb:250:in `records'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/bullet-7.0.1/lib/bullet/active_record60.rb:46:in `records'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/relation.rb:245:in `to_ary'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/associations/association.rb:205:in `find_target'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/associations/collection_association.rb:274:in `load_target'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/bullet-7.0.1/lib/bullet/active_record60.rb:176:in `load_target'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/associations/collection_proxy.rb:43:in `load_target'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/associations/collection_proxy.rb:1002:in `records'
/home/vagrant/foreman/.vendor/ruby/2.7.0/gems/activerecord-6.0.4.8/lib/active_record/relation/delegation.rb:88:in `each'
/home/vagrant/foreman/app/models/host/base.rb:562:in `detect'
/home/vagrant/foreman/app/models/host/base.rb:562:in `get_interface_by_flag'
/home/vagrant/foreman/app/models/host/base.rb:248:in `primary_interface'
/home/vagrant/foreman/app/models/host/base.rb:103:in `fqdn'
/home/vagrant/foreman_rh_cloud/lib/foreman_inventory_upload.rb:83:in `inventory_self_url'
/home/vagrant/foreman_rh_cloud/lib/inventory_sync/async/inventory_self_host_sync.rb:34:in `request_url'
/home/vagrant/foreman_rh_cloud/lib/inventory_sync/async/query_inventory_job.rb:52:in `query_inventory'
/home/vagrant/foreman_rh_cloud/lib/inventory_sync/async/query_inventory_job.rb:31:in `block (3 levels) in run'
/home/vagrant/foreman_rh_cloud/lib/inventory_sync/async/query_inventory_job.rb:30:in `loop'
/home/vagrant/foreman_rh_cloud/lib/inventory_sync/async/query_inventory_job.rb:30:in `block (2 levels) in run'
```